docs/side_quests/working_with_files.md

```diff
@@ -0,0 +1,998 @@
+            [
+                id: sample,
+                replicate: replicate.replace('rep', ''),
+                type: type,
+                readNum: readNum,
+            ],
+            fastqs
+        ]
+    }
+    .view()
+    ```
+
+=== "Before"
+
+    ```groovy title="main.nf" linenums="3" hl_lines="3-11"
+    ch_fastq = Channel.fromFilePairs('data/sampleA_rep1_normal_R{1,2}_001.fastq.gz')
+    .view()
+    ```
+
+```bash
+nextflow run main.nf
+```
+
+```console title="File Pairs Output parsed"
+
+ N E X T F L O W   ~  version 24.10.5
+
+Launching `main.nf` [prickly_stonebraker] DSL2 - revision: f62ab10a3f
+
+[[id:sampleA, replicate:1, type:normal, readNum:R], [/workspaces/training/side-quests/working_with_files/data/sampleA_rep1_normal_R1_001.fastq.gz, /workspaces/training/side-quests/working_with_files/data/sampleA_rep1_normal_R2_001.fastq.gz]]
+```
+
+Well done! We have grabbed the metadata from the filenames and used them as values in the tuple.
+
+### Takeaway
+
+- [`Channel.fromFilePairs()` automatically finds and pairs related files](https://www.nextflow.io/docs/latest/reference/channel.html#fromfilepairs)
+- The closure defines how to extract sample IDs from filenames
+- This simplifies handling paired-end reads in your pipeline
+
+---
+
+## 5. Using File Operations in Processes
+
+Now let's put all this together in a simple process to demonstrate how to use file operations in a Nextflow process.
+
+### 5.1. Create the process
+
+We'll keep it simple and make a process called `ANALYZE_READS` that takes in a tuple of metadata and a pair of fastq files and analyses them. We could imagine this is an alignment, or variant calling or any other step.
+
+Add the following to the top of your `main.nf` file:
+
+=== "After"
+
+    ```groovy title="main.nf - process example" linenums="1" hl_lines="1-23"
+    process ANALYZE_READS {
+        tag "${meta.id}"
+
+        publishDir "results/${meta.id}", mode: 'copy'
+
+        input:
+        tuple val(meta), path(fastqs)
+
+        output:
+        tuple val(meta.id), path("${meta.id}_stats.txt")
+
+        script:
+        """
+        echo "Sample metadata: ${meta.id}" > ${meta.id}_stats.txt
+        echo "Replicate: ${meta.replicate}" >> ${meta.id}_stats.txt
+        echo "Type: ${meta.type}" >> ${meta.id}_stats.txt
+        echo "Read 1: ${fastqs[0]}" >> ${meta.id}_stats.txt
+        echo "Read 2: ${fastqs[1]}" >> ${meta.id}_stats.txt
+        echo "File sizes:" >> ${meta.id}_stats.txt
+        echo "Read 1 size: \$(gunzip -dc ${fastqs[0]} | wc -l | awk '{print \$1/4}') reads" >> ${meta.id}_stats.txt
+        echo "Read 2 size: \$(gunzip -dc ${fastqs[1]} | wc -l | awk '{print \$1/4}') reads" >> ${meta.id}_stats.txt
+        """
+    }
+
+    workflow {
+    ```
+
+=== "Before"
+
+    ```groovy title="main.nf" linenums="1"
+    workflow {
+    ```
+
+!!! note
+
+    We are calling our map '`meta`'. This is the first introduction of a concept called `metamap` which we will cover later!
```

Comment:
I am not sure what we want to cover beyond this tbh. The next step would maybe be adding fields via nf-schema, but I think that would be a training focused on input validation